discovery-provider/src/eth_indexing/event_scanner_db_state.py

```diff
@@ -0,0 +1,171 @@
+                logger.info(
+                    f"Saving {len(transfer_events)} scanned eth transfer events to database"
+                )
+                session.bulk_save_objects(transfer_events)
+
+        """Save last scanned block to redis."""
+        logger.info(f"Saving last scanned block ({self.last_scanned_block}) to redis")
+        redis_set_and_dump(
+            self.redis,
+            eth_indexing_last_scanned_block_key,
+            str(self.last_scanned_block),
+        )
+
+        # reset state blocks and seconds since last save
+        self.blocks = dict()
+        self.seconds_since_last_save = time.time()
+
+    def get_last_scanned_block(self) -> int:
+        """The number of the last block we have stored."""
+        return self.last_scanned_block
+
+    def delete_potentially_forked_data(self, since_block):
+        """Remove potentially reorganised blocks from the scan data."""
+        # delete eth blocks from db using block_num and self.last_scanned_block in the query filter
+        pass
+
+    def start_chunk(self, block_number: int, chunk_size: int):
+        pass
+
+    def end_chunk(self, block_number: int):
+        """Save at the end of each block, so we can resume in the case of a crash or CTRL+C"""
+        # Next time the scanner is started we will resume from this block
+        self.last_scanned_block = block_number
+
+        if time.time() - self.seconds_since_last_save >= MIN_SECONDS_SINCE_LAST_SAVE:
+            self.save()
+
+    def process_event(self, block_when: datetime.datetime, event: TransferEvent) -> str:
+        """Record a ERC-20 transfer in our database."""
+        # Events are keyed by their transaction hash and log index
+        # One transaction may contain multiple events
+        # and each one of those gets their own log index
+
+        # event_name = event.event # "Transfer"
+        log_index = event["logIndex"]  # Log index within the block
+        # transaction_index = event.transactionIndex  # Transaction index within the block
+        txhash = event["transactionHash"].hex()  # Transaction hash
+        block_number = event["blockNumber"]
+        # SHOULD WE ALSO INCLUDE BLOCKHASH HERE?
+
+        # Convert ERC-20 Transfer event to our internal format
+        args = event["args"]
+        transfer = {
+            "from": args["from"],
+            "to": args["to"],
+            "value": args["value"]
+            / 1e18,  # divide by 10^18 to get correct transfer value
+            "timestamp": block_when,
+        }
+
+        # add user ids from the transfer event into the balance refresh queue
+        transfer_event_wallets = [transfer["from"].lower(), transfer["to"].lower()]
+        with self.db.scoped_session() as session:
+            result = (
+                session.query(User.user_id)
+                .outerjoin(AssociatedWallet, User.user_id == AssociatedWallet.user_id)
+                .filter(User.is_current == True)
```

Comment:
I think you need to also filter on `AssociatedWallet.is_current == True` and `AssociatedWallet.is_delete != True`

Reply:
Good catch